### ISSUE TITLE
Forge version

### ISSUE THREAD

**Issue by KitsuneNoNatsu** [tr7zw/ItemSwapper#25]
Will a forge version ever be released, it would be really helpful for modpack and mod developers to introduce item groups and modify item groups

**Comment by tr7zw** [comment]
For now not planned, because Forge has too many limitations for the resource pack part(and just complicated the current development process). The backend is still there, so might do that sometime later, when the mod is more finished(but with the limitations that result in using Forge).

**Comment by KitsuneNoNatsu** [comment]
I know it would require a total rewrite, but why not make this for Acrhitectury API?
I'm not too knowledgeable in how mod development works, but is there a reason this API was not chosen?

**Comment by tr7zw** [comment]
No, it doesn't require a total rewrite or the Architectury API. 
As you might have noticed, there is even a Forge project in this repo, 
![grafik](https://user-images.githubusercontent.com/5133936/206917375-d1026b9d-fd41-4963-b427-7a160c768b0b.png). 
The thing is, Forge would just slow down development since you have to always also check that nothing broke there again, I'd have to figure out how to work around Forge's messy limitations(1.19.3 Forge somehow making it worse) and personally, I don't like or care about Forge. Everyone had years now to switch to Fabric, and when people are still using this slow and old loader, they will get the Forge port of this mod also slower and later.
Yes, there probably will be a port later, but only when the mod is nearing completion. If you want it now, instead consider giving Fabric/Quilt a shot, having more mods, fewer bugs, faster development, and better tooling.

**Comment by tr7zw** [comment]
To quote another developer that is fed up with Forge still being a thing:
![https://images-ext-2.discordapp.net/external/x2G8b1gtQbq1mlH05NbqplyNmPkRpjhkfBtRFKcheBc/https/cammiescorner.dev/images/extras/no_forge.png](https://images-ext-2.discordapp.net/external/x2G8b1gtQbq1mlH05NbqplyNmPkRpjhkfBtRFKcheBc/https/cammiescorner.dev/images/extras/no_forge.png)

**Comment by KitsuneNoNatsu** [comment]
> No, it doesn't require a total rewrite or the Architectury API. As you might have noticed, there is even a Forge project in this repo, ![grafik](https://user-images.githubusercontent.com/5133936/206917375-d1026b9d-fd41-4963-b427-7a160c768b0b.png). The thing is, Forge would just slow down development since you have to always also check that nothing broke there again, I'd have to figure out how to work around Forge's messy limitations(1.19.3 Forge somehow making it worse) and personally, I don't like or care about Forge. Everyone had years now to switch to Fabric, and when people are still using this slow and old loader, they will get the Forge port of this mod also slower and later. Yes, there probably will be a port later, but only when the mod is nearing completion. If you want it now, instead consider giving Fabric/Quilt a shot, having more mods, fewer bugs, faster development, and better tooling.

That sounds great actually
I completely agree with the sentiment, I'm also tired of Forge being the "default" for people, where Fabric is better in every single regard
I was asking on behalf of people less knowledgeable, and thank you for clearing it up
Looking forward to all the future development of this project, because it is amazing

**Comment by tr7zw** [comment]
Yea the issue can stay open for future reference, and as I said, a Forge version will be made... later.

**Comment by ChaosDogG** [comment]
so u've openly stated how stinky Forge is...but wut of NeoForge? from my knowledge, NF is trying to make itself a better version of Forge, and perhaps even outright become the one true modloader in the (perhaps distant) future like Forge once was before 1.13

**Comment by tr7zw** [comment]
Making a Forge version = also getting a NeoForge version. As for everything codewise they are (mostly) the same. 
But in general, I don't use Forge/NeoForge, so I personally don't care too much about them. During the development of this mod, I didn't want to bother with having to maintain them as it would just slow stuff down. Now I have the dev tools setup to build 1.16.5-1.20.6 fabric/forge/neoforge from the same codebase, but just haven't had the time/interest to look into all the weird stuff about the Forges like built-in Resourcepacks, networking etc.
This mod is also open source, if anyone really wants a Forge/NeoForge version, nothing is stopping anyone from adding it to the build, pr'ing it and then I have no problem maintaining that.
